airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.test.ts

```diff
@@ -654,4 +674,128 @@ describe("calculateInitialCatalog", () => {
+              ...config,
+              destinationSyncMode: DestinationSyncMode.append_dedup,
+              syncMode: SyncMode.incremental,
+              primaryKey: [["id"]],
+            },
+          },
+        ],
+      },
+      [DestinationSyncMode.append_dedup],
+      [
+        {
+          transformType: StreamTransformTransformType.update_stream,
+          streamDescriptor: { name: "test", namespace: "namespace-test" },
+          updateStream: [
+            {
+              breaking: true,
+              transformType: FieldTransformTransformType.remove_field,
+              fieldName: ["id"],
+            },
+          ],
+        },
+      ],
+      false
+    );
+    expect(values.streams[0].config?.primaryKey).toEqual([]);
+    expect(values.streams[1].config?.primaryKey).toEqual([["id"]]);
+  });
+  it("should remove cursor from config if the field was removed during refresh", () => {
+    const { config, stream: sourceDefinedStream } = mockSyncSchemaStream;
+    const values = calculateInitialCatalog(
+      {
+        streams: [
+          {
+            id: "1",
+            stream: {
+              ...sourceDefinedStream,
+              name: "test",
+              sourceDefinedCursor: false,
+              defaultCursorField: [],
+              sourceDefinedPrimaryKey: [],
+              supportedSyncModes: [SyncMode.incremental],
+            },
+            config: {
+              ...config,
+              destinationSyncMode: DestinationSyncMode.append_dedup,
+              syncMode: SyncMode.incremental,
+              cursorField: ["updated_at"],
+            },
+          },
+          {
+            id: "2",
+            stream: {
+              ...sourceDefinedStream,
+              name: "test-2",
+              sourceDefinedCursor: true,
+              defaultCursorField: ["updated_at"],
+              supportedSyncModes: [SyncMode.incremental],
+            },
+            config: {
+              ...config,
+              destinationSyncMode: DestinationSyncMode.append_dedup,
+              syncMode: SyncMode.incremental,
+              primaryKey: [["id"]],
+            },
+          },
+        ],
+      },
+      [DestinationSyncMode.append_dedup],
+      [
+        {
+          transformType: StreamTransformTransformType.update_stream,
+          streamDescriptor: { name: "test", namespace: "namespace-test" },
+          updateStream: [
+            {
+              breaking: true,
+              transformType: FieldTransformTransformType.remove_field,
+              fieldName: ["updated_at"],
+            },
+          ],
+        },
+      ],
+      false
+    );
+    expect(values.streams[0].config?.cursorField).toEqual([]);
+    expect(values.streams[1].config?.primaryKey).toEqual([["id"]]);
+  });
+
+  it("should not populate the source default cursor if the user-specified cursor field was removed during refresh", () => {
+    expect(2 + 2).toEqual(4);
+    // tbd expected behavior...
```

Comment:
https://airbytehq-team.slack.com/archives/C042ABK94FR/p1670451834601009?thread_ts=1670360753.045569&cid=C042ABK94FR

Waiting for feedback on what the desired behavior is.  As it stands, if there is a source default cursor (but it isn't a source-defined cursor), we will populate it since the old cursor was removed and the property is now empty...